pkg/cmd/roachtest/operations/node_kill.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package operations
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/operation"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/option"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/registry"
+	"github.com/cockroachdb/cockroach/pkg/roachprod"
+	"github.com/cockroachdb/cockroach/pkg/util/randutil"
+)
+
+type cleanupNodeKill struct {
+	nodes option.NodeListOption
+}
+
+func (cl *cleanupNodeKill) Cleanup(ctx context.Context, o operation.Operation, c cluster.Cluster) {
+	// We might need to restart the node if it isn't live.
+	db, err := c.ConnE(ctx, o.L(), cl.nodes[0])
+	if err != nil {
+		c.Run(ctx, option.WithNodes(cl.nodes), "./cockroach.sh")
+		return
+	}
+	defer db.Close()
+	_, err = db.Query("SELECT 1")
+	if err != nil {
+		c.Run(ctx, option.WithNodes(cl.nodes), "./cockroach.sh")
+	}
+}
+
+func nodeKillRunner(signal int, drain bool) func(ctx context.Context, o operation.Operation, c cluster.Cluster) registry.OperationCleanup {
+	return func(ctx context.Context, o operation.Operation, c cluster.Cluster) registry.OperationCleanup {
+		return runNodeKill(ctx, o, c, signal, drain)
+	}
+}
+
+func runNodeKill(
+	ctx context.Context, o operation.Operation, c cluster.Cluster, signal int, drain bool,
+) registry.OperationCleanup {
+	rng, _ := randutil.NewPseudoRand()
+
+	nodes := c.All()
+	nid := nodes[rng.Intn(len(nodes))]
+
+	if drain {
+		o.Status(fmt.Sprintf("draining node %d", nid))
+
+		url, err := c.ExternalPGUrl(ctx, o.L(), c.Node(nid), roachprod.PGURLOptions{
+			External: false,
+			Secure:   c.IsSecure(),
+		})
+		if err != nil {
+			o.Fatal(err)
+		}
+		err = c.RunE(ctx, option.WithNodes(c.Node(nid)),
```

Comment:
@renatolabs may know how to structure this better, but we could instead of doing an `ExternalPGUrl` use the new command util to optionally add `insecure` or `certs` to `drain`:

Something like this (untested):
```
		cmd := roachtestutil.NewCommand(fmt.Sprintf("%s node drain", test.DefaultCockroachPath)).
				Flag("logtostderr", "INFO").
				Flag("port", fmt.Sprintf("{pgport:%d}", nid)).
				MaybeOption(!c.IsSecure(),"insecure").
				MaybeFlag(c.IsSecure(),"certs", install.CockroachNodeCertsDir).
				Option("self")
		err = c.RunE(ctx, option.WithNodes(c.Node(nid)), cmd.String())
```